File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b){
            return b[0] > a[0];
        });
        res.push_back(intervals[0]);
        int l = 0;
        for(int r = 1; r < intervals.size(); r++)
        {
            if((res[l][1] >= intervals[r][0] && res[l][1] <= intervals[r][1]) || (intervals[r][1] >= res[l][0] && intervals[r][1] <= res[l][1]))
            {
                res[l][1] = max(intervals[r][1], res[l][1]);
            }
            else
            {
                res.push_back(intervals[r]);
                l++;
            }
        }
        return res;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res = []
        intervals.sort()
        curS = intervals[0][0]
        curE = intervals[0][1]
        for s, e in intervals[1:]:
            if (s >= curS and s <= curE) or (curE >= s and curE <= e):
                curS = min(s, curS)
                curE = max(e, curE)
            else:
                res.append([curS, curE])
                curS = s
                curE = e
        res.append([curS, curE])
        return res
